src/core.js

```diff
@@ -295,6 +295,23 @@ class Kuroshiro {
             }
         }
     }
+
+    /**
+     * Asynchronously convert given string to target syllabary with options available
+     * @memberOf Kuroshiro
+     * @instance
+     * @param {string} str Given String
+     * @param {Object} [options] Settings Object
+     * @param {string} [options.to="hiragana"] Target syllabary ["hiragana"|"katakana"|"romaji"]
+     * @param {string} [options.mode="normal"] Convert mode ["normal"|"spaced"|"okurigana"|"furigana"]
+     * @param {string} [options.romajiSystem="hepburn"] Romanization System ["nippon"|"passport"|"hepburn"]
+     * @param {string} [options.delimiter_start="("] Delimiter(Start)
+     * @param {string} [options.delimiter_end=")"] Delimiter(End)
+     * @returns {Promise<string>} Promise result of conversion
+     */
+    convert(str, options) {
+        return Promise.resolve(this.convertSync(str, options));
```

Comment:
I believe this should be the following:
```suggestion
        return Promise.resolve().then(() => this.convertSync(str, options));
```

This is based on what `import(...)` is transpiled to when targeting CommonJS, and this approach has a few benefits:

- The execution is delayed instead of done immediately (better async I/O).
- If `this.convertSync(str, options)` throws, this will cause a promise rejection, currently it throws a sync error, which makes `.catch()` not work.
- Much cleaner than a try/catch using Promise.resolve and Promise.reject.